### ISSUE TITLE
`]%` ignores some tokens

### ISSUE THREAD

**Issue by lacygoill** [chrisbra/matchit#1]
Hello,

I open this issue because you asked to open issues for the various problems discussed on this question on [stackexchange](https://vi.stackexchange.com/questions/13155/what-do-matchit-vims-a-do/13593?noredirect=1#comment23295_13593).

Currently, it seems that `]%` only moves the cursor on a closed parenthesis, and ignores all tokens defined in `b:match_words`. The issue may come from the [line 695](https://github.com/chrisbra/matchit/blob/90f0ee7d0f3f997745c87ad2a49bdde3e3cb47d1/plugin/matchit.vim#L695):

     let closepat = substitute(closepat, ',', '\\|', 'g')

The last step producing the pattern describing the end token involves a substitution which replaces the commas (separating the groups of tokens inside `b:match_words`) with alternations, but not the colons (separating the tokens inside a group). So, maybe this line would work better:

    let closepat = substitute(closepat, '[,:]', '\\|', 'g')

I won't open an issue for the `[%` motion, because it seems you have already fixed it in these 2 commits:

https://github.com/chrisbra/matchit/commit/68b2f2224d8a13eb2a32e543d9847a4fc364e619
https://github.com/chrisbra/matchit/commit/ecff10ab2bb4fba4f1926d0e7716b930d9f00b80

**Comment by chrisbra** [comment]
Hi, I am holding back, until it is clear, which is the final upstream repository for the matchit plugin. See benjifisher/matchit.zip#10

**Comment by k-takata** [comment]
I created a fix for this in my [repository](https://github.com/k-takata/matchit/tree/devel), and I'm going to create some PRs after #4 and #5 are merged.
However, I'm wondering how `]%` should work for multiple patterns like `\<if\>:\<else\>:\<endif\>`.

```
if
  | <- assuming the cursor is here.
else
endif
```

If `]%` is pressed, which should the cursor go to? `else` or `endif`?
With my current implementation, it goes to `else`.

**Comment by lacygoill** [comment]
@k-takata In this scenario, I would expect the cursor to move to `else`.

**Comment by chrisbra** [comment]
Hi @k-takata  if you like to have your fixes merged, you can now create a new PR then.

**Comment by k-takata** [comment]
I think this is fixed with #8.
